src/utils/typedData.ts

```diff
@@ -107,130 +174,218 @@ function getMerkleTreeType(types: TypedData['types'], ctx: Context) {
 /**
  * Encode a type to a string. All dependent types are alphabetically sorted.
  */
-export const encodeType = (types: TypedData['types'], type: string): string => {
-  const [primary, ...dependencies] = getDependencies(types, type);
+export function encodeType(
+  types: TypedData['types'],
+  type: string,
+  revision: Revision = Revision.Legacy
+): string {
+  const [primary, ...dependencies] = getDependencies(types, type, undefined, undefined, revision);
   const newTypes = !primary ? [] : [primary, ...dependencies.sort()];
 
+  const esc = revisionConfiguration[revision].escapeTypeString;
+
   return newTypes
     .map((dependency) => {
-      return `${dependency}(${types[dependency].map((t) => `${t.name}:${t.type}`)})`;
+      const dependencyElements = types[dependency].map((t) => {
+        const targetType =
+          t.type === 'enum' && revision === Revision.Active
+            ? (t as StarkNetEnumType).contains
+            : t.type;
+        // parentheses handling for enum variant types
+        const typeString = targetType.match(/^\(.*\)$/)
+          ? `(${targetType
+              .slice(1, -1)
+              .split(',')
+              .map((e) => (e ? esc(e.trim()) : e))
```

Comment:
i wouldn't trim trim it, regarding the spec `Thing` and `Thing ` are two separate types.
We could choose to reject request that are not trimmed but automatically trimming might be dangerous

Reply:
Good point, I added the trim so the `(u128, u128*) `example from the SNIP could pass. 

I removed the trimming for this PR, it would probably be best to re-add it after updating the SNIP so whitespace characters are excluded from string borders like the parenthesis rule. It would also make sense to exclude commas from the allowed characters for custom type names since they are used as enum delimiters.